File: leetcode/daily-code/10-26.cpp
class Solution {
public:
    vector<int> smallerNumbersThanCurrent(vector<int>& nums) {
        vector<int> ret;
        int count[101] = {0};
        for(auto v :nums){
            count[v]++;
        }
        for(int i = 1 ; i < 101; ++i){
            count[i] += count[i-1];
        }
        for(auto v:nums){
            ret.push_back(v?count[v-1]:0);
        }
        return ret;
    }
};

File: leetcode/daily-code/10-26.js
var smallerNumbersThanCurrent = function(nums) {
    const cnt = new Array(101).fill(0);
    const n = nums.length;
    for(let i = 0; i<n ; ++i){
        cnt[nums[i]] +=1;
    }

    for(let i = 0; i<=100;++i){
        cnt[i]+= cnt[i-1];
    }

    const ret = [];
    for(let i = 0; i < n; ++i){
        ret.push(nums[i] ? cnt[nums[i] - 1]:0);
    }
    return ret;

}
